src/approaches/nsrt_rl_approach.py

```diff
@@ -72,10 +90,128 @@ def learn_from_interaction_results(
                                       actions,
                                       _is_demo=False,
                                       _train_task_idx=i)
-            self._online_trajectories.append(traj)
+            self._train_task_to_online_traj[i] = traj
+
+
+        option_to_data = {} # option_name -> online experience data
+
+        # For each task, for each _Option involved in the trajectory, compute
+        # and store (s, a, s', r, relative_param) data.
+        for i in range(len(self._train_tasks)):
+            plan = self._train_task_to_option_plan[i]
+            traj = self._train_task_to_online_traj[i]
+
+            curr_option_idx = 0
+            curr_option = plan[curr_option_idx]
+            if curr_option.name not in option_to_data:
+                option_to_data[curr_option.name] = []
+            curr_states = [traj.states[0]]
+            curr_actions = []
+            curr_rewards = []
+            curr_relative_params = []
+            actions = (a for a in traj.actions)
+            block = [b for b in curr_option.objects if b.type.name=='block'][0]
+            robot = [r for r in curr_option.objects if r.type.name=='robot'][0]
+
+            # Generate transition data, (s, a, s', r, relative_param). The
+            # reward R(s, a, s') = neg_reward if s' is not within epsilon of the
+            # subgoal, and pos_reward otherwise.
+            for j, s in enumerate(traj.states[1:]):
+                curr_states.append(s)
+                curr_actions.append(next(actions))
+                # TODO: is there a better way to do this based on
```

Comment:
yeah I see the issue. here are a few possibilities:
1. we could store the relevant information in the option's memory. that would involve a change in `_LearnedNeuralParameterizedOption._precondition_based_initiable`.
2. we could create a subclass of `_Option` that exposes the relevant information with a method.
3. we could refactor the code so that when the relevant information is computed the first time, it's passed back to this Approach, which then saves it.

I think (3) is preferred if we can figure out a way to do that cleanly, but I can't immediately envision such a way. absent that, (1) is probably preferred.

Reply:
maybe we can achieve (3) by using the new option learner class that you're introducing?

Reply:
I think the cleanest way, without making any modifications, would use the `self._changing_var_to_feat` attribute of the `_LearnedNeuralParamterizedOption`. But for some reason, that's the one attribute I couldn't access– it just didn't exist when I listed the attributes (via `o.dir()`, where o is the `_Option`. What confused me was that I did see `self._changing_var_order`. We can discuss during our meeting today.